### ISSUE TITLE
Interruption of jsonrpc communication results in watchdog timer infinitely triggering

### ISSUE THREAD

**Issue by ksunden** [bluesky/bluesky-queueserver#242]
Initially appears as unexpected JSON-RPC messages being processed, but escalates to an exception in handling of JSON-RPC and then the watchdog timer going haywire.

## Expected Behavior
I expect my plan to continue executing.

## Current Behavior
Plan stops executing after watchdog timer continuously triggers

## Possible Solution
I suspect it is async usage of jsonrpc causing race conditions

Possibly having a list of expected ids rather than a single id could help, but not sure, haven't dug into the implementation enough to know

## Steps to Reproduce (for bugs)
Unfortunately this seems to be a stochastic race condition that I've only ever seen after running for hours (usually around 12-16+ hours into a queue)... so it falls into the nasty region of very difficult to pin down causes but problematic enough to be a showstopping bug when doing overnight data acquisition.

## Context

Timeline starts with data writer process going dark (it was completely down when I got in today) It seems to have successfully gotten the data and started processing the stop document of the last successful scan, but did not produce mpl images nor provide any useful logs...This one may be a bit of a dead end for right now, but providing context

The run engine went on to the next scan, and around 420 points in got the following error message:

```
2022-06-21T12:01:45.552057800Z |      416 | 07:01:45.5 |   4566.210 |   2493.766 |     -0.000 |      0.001 |     -0.002 |      1.687 |     -0.000 |      0.000 |      0.000 |     -0.000 |             0.000 |           0.000 |          -0.000 |     -1.000 |    504.022 |     -0.500 |
[E 2022-06-21 07:01:47,991 bluesky_queueserver.manager.comms] Unsolicited message received: {'id': 'f9e0de27-7bc2-4100-9953-4117db352dd3',
 'jsonrpc': '2.0',
 'result': {'background_tasks_num': 0,
            'completed_tasks_available': False,
            'environment_state': 'executing_plan',
            'plans_and_devices_list_updated': False,
            're_deferred_pause_requested': False,
            're_report_available': False,
            're_state': 'running',
            'run_list_updated': False,
            'running_item_uid': 'd1667007-e546-41ff-8bdf-990ffbfea803',
            'running_plan_completed': False,
            'running_task_uid': None}}. Message is ignored
```
And kept going for about 15 more points before getting a second similar message followed by an asyncio traceback:
```
2022-06-21T12:02:17.754249000Z |      429 | 07:02:17.7 |   4587.156 |   2227.171 |     -0.000 |     -0.001 |     -0.002 |      1.689 |     -0.000 |     -0.000 |      0.000 |      0.000 |            -0.000 |          -0.000 |           0.000 |     -1.000 |    491.885 |     -0.500 |
[E 2022-06-21 07:02:21,416 bluesky_queueserver.manager.comms] Unsolicited message received: {'id': 'ddb8ed41-009b-4100-8e4a-cf9f3b40107e',
 'jsonrpc': '2.0',
 'result': {'background_tasks_num': 0,
            'completed_tasks_available': False,
            'environment_state': 'executing_plan',
            'plans_and_devices_list_updated': False,
            're_deferred_pause_requested': False,
            're_report_available': False,
            're_state': 'running',
            'run_list_updated': False,
            'running_item_uid': 'd1667007-e546-41ff-8bdf-990ffbfea803',
            'running_plan_completed': False,
            'running_task_uid': None}}. Message is ignored
ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-142123' coro=<PipeJsonRpcSendAsync._response_received() done, defined at /usr/local/lib/python3.9/dist-packages/bluesky_queueserver/manager/comms.py:383> exception=InvalidStateError('invalid state')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/bluesky_queueserver/manager/comms.py", line 400, in _response_received
    self._fut_comm.set_result(response)
asyncio.exceptions.InvalidStateError: invalid state
2022-06-21T12:02:33.215901400Z |      430 | 07:02:33.1 |   4587.156 |   2232.143 |      0.000 |      0.001 |     -0.001 |      1.683 |      0.000 |      0.000 |     -0.000 |     -0.000 |             0.000 |          -0.000 |          -0.000 |     -1.000 |    492.128 |     -0.500 |
```
Then it elevated from "Unsolicited message/message ignored" to still the same contents but now "incorrect message ID"
```
2022-06-21T12:02:58.349957200Z |      441 | 07:02:58.3 |   4587.156 |   2288.330 |      0.000 |      0.003 |      0.000 |      1.674 |      0.000 |     -0.000 |      0.000 |      0.000 |             0.000 |          -0.000 |          -0.000 |     -1.000 |    494.808 |     -0.500 |
[E 2022-06-21 07:03:03,893 bluesky_queueserver.manager.comms] Received response with incorrect message ID: f0bbf483-b9ef-48c6-8b0c-681c0a63697a. Expected 8da292c6-cb87-4aab-95d8-e0f720775034.
Message: {'id': 'f0bbf483-b9ef-48c6-8b0c-681c0a63697a',
 'jsonrpc': '2.0',
 'result': {'background_tasks_num': 0,
            'completed_tasks_available': False,
            'environment_state': 'executing_plan',
            'plans_and_devices_list_updated': False,
            're_deferred_pause_requested': False,
            're_report_available': False,
            're_state': 'running',
            'run_list_updated': False,
            'running_item_uid': 'd1667007-e546-41ff-8bdf-990ffbfea803',
            'running_plan_completed': False,
            'running_task_uid': None}}
2022-06-21T12:03:04.888053100Z |      442 | 07:03:04.8 |   4587.156 |   2293.578 |     -0.000 |      0.000 |      0.001 |      1.690 |     -0.000 |     -0.000 |     -0.000 |      0.000 |             0.000 |          -0.000 |           0.000 |     -1.000 |    495.051 |     -0.500 |
```
The expected message ID does not match any prior unsolicited message

Then about 40 seconds later it started getting watchdog timer consistently triggering with data points still trickling in:

```
2022-06-21T12:03:38.867002800Z |      453 | 07:03:38.8 |   4587.156 |   2352.941 |      0.000 |     -0.003 |      0.001 |      1.677 |      0.000 |     -0.000 |     -0.000 |      0.000 |            -0.000 |          -0.000 |           0.000 |     -1.000 |    497.762 |     -0.500 |
[E 2022-06-21 07:03:45,743 bluesky_queueserver.manager.start_manager] Timeout detected by Watchdog. RE Manager malfunctioned and must be restarted
[I 2022-06-21 07:03:45,961 bluesky_queueserver.manager.manager] Starting ZMQ server at 'tcp://*:60615'
[I 2022-06-21 07:03:45,970 bluesky_queueserver.manager.manager] ZMQ control channels: encryption disabled
2022-06-21T12:03:46.083414900Z |      454 | 07:03:46.0 |   4587.156 |   2358.491 |      0.000 |      0.003 |     -0.000 |      1.684 |      0.000 |     -0.000 |     -0.000 |      0.000 |            -0.000 |          -0.000 |           0.000 |     -1.000 |    498.009 |     -0.500 |
[I 2022-06-21 07:03:46,080 bluesky_queueserver.manager.manager] Starting RE Manager process
[I 2022-06-21 07:03:48,477 bluesky_queueserver.manager.manager] Downloading the lists of existing plans and devices from the worker environment
[I 2022-06-21 07:03:48,561 bluesky_queueserver.manager.manager] Starting ZeroMQ server ...
[I 2022-06-21 07:03:48,666 bluesky_queueserver.manager.manager] ZeroMQ server is waiting on tcp://*:60615
[I 2022-06-21 07:03:49,179 bluesky_queueserver.manager.manager] Returning plan history ...
2022-06-21T12:03:49.403787700Z |      455 | 07:03:49.3 |   4587.156 |   2364.066 |     -0.000 |      0.001 |      0.002 |      1.683 |      0.000 |     -0.000 |     -0.000 |     -0.000 |             0.000 |          -0.000 |          -0.000 |     -1.000 |    498.261 |     -0.500 |
[I 2022-06-21 07:03:49,505 bluesky_queueserver.manager.manager] Returning current queue and running plan ...
2022-06-21T12:03:53.622222600Z |      456 | 07:03:53.5 |   4587.156 |   2369.668 |      0.000 |      0.003 |     -0.002 |      1.672 |     -0.000 |      0.000 |     -0.000 |      0.000 |            -0.000 |           0.000 |           0.000 |     -1.000 |    498.504 |     -0.500 |
```
The watchdog timer continues to trip until we see the last data point collected (472 out of an expected 1891, so 25% in)And the scan indicates failed exit status with the following stop document:
```
('stop', Stop({'run_start': '6dc0a157-9b5f-4e16-b855-cee1653578fa', 'time': 1655817508.7386343, 'uid': '49a3f4da-5164-4d3a-97f8-7079e41d2561', 'exit_status': 'fail', 'reason': "A second 'create' message is not allowed until the current event bundle is closed with a 'save' or 'drop' message.", 'num_events': {'baseline': 1, 'primary': 472}}))
```
## Your Environment

We use a set of docker containers to run all of the queueserver services together, on a Windows host:[github.com/wright-group/bluesky-in-a-box](http://github.com/wright-group/bluesky-in-a-box)This consists of 6 containers:

  -  re-manager: main queueserver container
  -  hwproxy: used to get readonly access to hardware for building UIs
  -  zmq-proxy: to forward event streams to a data writing container (wt5) plus our live plots in our UIs... maybe should be replaced by kafka but it gets the job done
  -  redis
  -  mongo
  -  wt5: data file writing to hdf5 files my scientists are used to and automatic mpl plots

**Comment by ksunden** [comment]
Seems likely that this behavior all stemmed from memory pressure rather than being a direct cause->effect like it seemed to me at first, so combined with #244 and fixing memory usage on my end I think I'm inclined to say "maybe it could be made even better but there is not a real problem in normal use, so not a priority" and close this.
